### PR TITLE
Cursed Russian Revolver

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Losing with the revolver turns you into a shade/soulstones you

Adminspawn/future exploration loot.

### CODE REVIEW

**Comment by Buggy123** [comment]
Interesting but I can't see how soulstoning yourself is a good reward for a away mission. I guess maybe if you can hold it to someone else's head and do it...

**Comment by KorPhaeron** [comment]
I don't get to steal the others soul when he loses with a normal one

File: code/game/gamemodes/wizard/soulstone.dm [hunk 1]
```diff
@@ -224,18 +224,19 @@
...
-	if(iscultist(U))
+	if(U)
+		S.faction |= "\ref[U]" //Add the master as a faction, allowing inter-mob cooperation
+	if(U && iscultist(U))
 		ticker.mode.add_cultist(S.mind,2)
 	S.cancel_camera()
 	C.icon_state = "soulstone2"
 	C.name = "Soul Stone: [S.real_name]"
-	if(iswizard(U) || usability)
+	if(U && iswizard(U) || usability)
```
Comment (tkdrg) [inline code/game/gamemodes/wizard/soulstone.dm]: needs parenthesis here
`if(U && (iswizard(U) || usability))`

**Comment by KorPhaeron** [comment]
@tkdrg 

Fixed

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 2]
```diff
@@ -251,11 +251,26 @@
...
+	name = "cursed russian revolver"
+	desc = "To play with this revolver requires wagering your very soul."
+
+/obj/item/weapon/gun/projectile/revolver/russian/soul/shoot_self(mob/living/user)
+	..()
+	var/obj/item/device/soulstone/anybody/SS = new /obj/item/device/soulstone/anybody(get_turf(src))
+	if(!SS.transfer_soul("FORCE", user)) //Something went wrong
+		qdel(SS)
+		return
+	user.visible_message("<span class='danger'>[user.name]'s soul is captured by \the [src]!</span>", "<span class='userdanger'>You've lost the gamble! Your soul is forfiet!</span>")
```
Comment (octareenroon91) [inline code/modules/projectiles/guns/projectile/revolver.dm]: forfeit

**Pull request merged**